packages/web/src/store/slices/websiteResults.ts

```diff
@@ -0,0 +1,75 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { WebPage, WebPagePackage, PackageVulnerabilityData } from '@gradejs-public/shared';
+import { DefaultFiltersAndSorters, FiltersState } from '../../components/layouts/Filters/Filters';
+import { client } from '../../services/apiClient';
+import { trackCustomEvent } from '../../services/analytics';
+
+const defaultDetectionResult = {
+  packages: [] as WebPagePackage[],
+  vulnerabilities: {} as Record<string, PackageVulnerabilityData[]>,
+  webpages: [] as WebPage[],
+};
+
+const initialState = {
+  filters: { ...DefaultFiltersAndSorters },
+  isFailed: false,
+  isLoading: false,
+  detectionResult: defaultDetectionResult,
+};
+
+const sleep = (ms: number | undefined) =>
+  new Promise((r) => {
+    setTimeout(r, ms);
+  });
+
+const hasPendingPages = (result: DetectionResult) =>
+  !!result.webpages.find((item) => item.status === 'pending');
+
+const getWebsite = createAsyncThunk('websiteResults/getWebsite', async (hostname: string) => {
+  const loadStartTime = Date.now();
+  await client.mutation('syncWebsite', hostname);
+  let results = await client.query('getWebsite', hostname);
+  while (hasPendingPages(results)) {
```

Comment:
What if a user loads a pending page and then goes on another page with results without waiting for the first page?

Reply:
Valid point! Added cleanup code in the component, though we should test it once again after redesign, because currently we don't have any way to get out of results page without page reload.